compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -535,7 +543,8 @@ impl<'f> PerFunctionContext<'f> {
                     let result = self.inserter.function.dfg.instruction_results(instruction)[0];
                     let array = *array;
 
-                    let expression = Expression::ArrayElement(array);
+                    let expression = references.expressions.get(&array).copied();
+                    let expression = expression.unwrap_or(Expression::ArrayElement(array));
```

Comment:
Same here. If you want we can keep these though since they're removed in the container refactor anyway.

Reply:
Done!